test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/crypto/HapiCryptoTransfer.java

```diff
@@ -258,11 +266,20 @@ private void misconfigureIfRequested(CryptoTransferTransactionBody.Builder b, Ha
 
 	@Override
 	protected long feeFor(HapiApiSpec spec, Transaction txn, int numPayerKeys) throws Throwable {
-		return spec.fees().forActivityBasedOp(
-				HederaFunctionality.CryptoTransfer,
-				(_txn, _svo) -> usageEstimate(_txn, _svo, spec.fees().tokenTransferUsageMultiplier()),
-				txn,
-				numPayerKeys);
+		if (feesObserver.isPresent()) {
+			return spec.fees().forActivityBasedOpWithDetails(
+					HederaFunctionality.CryptoTransfer,
+					(_txn, _svo) -> usageEstimate(_txn, _svo, spec.fees().tokenTransferUsageMultiplier()),
+					txn,
+					numPayerKeys,
+					feesObserver.get());
+		} else {
```

Comment:
nit : `else` is not required, can return directly

Reply:
Removed `else`.